### DISCUSSION TITLE
A few notes on GSOC 2024 Project 5: Better interfacing of Blender and MDAnalysis

### DISCUSSION THREAD

**Discussion by yuxuanzhuang** [MDAnalysis/mdanalysis#4480] [GSoC Q&A]
Regarding [GSoC-2024-Project-Ideas#project-5-better-interfacing-of-blender-and-mdanalysis](https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-2024-Project-Ideas#project-5-better-interfacing-of-blender-and-mdanalysis)

We recently conducted a workshop on MolecularNodes and MDAnalysis (https://github.com/MDAnalysis/MDAnalysisWorkshop-Intro0.5Day/tree/feb24-ws/blender). Try installing it and experimenting with it to become familiar with the current interface. Consider what you expect to visualize that is currently missing and how the interface could be improved.

**A few useful links**

- Check the current MDAnalysis parser in MolecularNodes (https://github.com/BradyAJohnston/MolecularNodes/blob/main/molecularnodes/io/parse/mda.py).

- The method of displaying interatomic distances in https://github.com/MDAnalysis/MDAnalysisWorkshop-Intro0.5Day/blob/feb24-ws/blender/Interactive_Blender_Notebook.ipynb is very hard-coded. Consider whether it could be extended to a more general BaseClass that could be inherited for broader usage.
- 
Please note that our goal is not to modify the existing Blender API, but rather to focus on improving the MolecularNodes module, either within the module itself or as a standalone package to visualize analysis results.

**Comment by Alentraca** [comment]
Hi,
I am Aligbeh Jahswill, a Graduate research candidate.
Delighted to have an introduction in participating as a Contributor to the MDAnalysis project. And particularly interested in the development of improved interfacing of Blender and MDAnalysis during the period of Google Summer of Code (GSoC).

Been familiar with MDAnalysis, also a participant of the workshop on MolecularNodes and MDAnalysis. It was great session with building on MDAnalysis and Fundamentals of Molecular Nodes.
(Thanks @BradyAJohnston and @yuxuanzhuang )

Eager to develop/improve module features and functionalities for visualization and analysis results in MDAnalysis with Blender.

I have considered visualizing(and detecting) conformational changes, visualizing secondary structure elements, and displaying distance between atomgroups (as exemplified in the workshop)

Also with regards to the method of displaying interatomic distances in https://github.com/MDAnalysis/MDAnalysisWorkshop-Intro0.5Day/blob/feb24-ws/blender/Interactive_Blender_Notebook.ipynb, I propose implementing a prototype function, such that it can be directly called into use [for example, simply importing the function before running the last 2 cells. and obtaining the Blender output]

Looking forward developing a more user-friendly and interactive visualization tool for analysis results in MDAnalysis, and receiving your guidance on taking next steps and what more I am to consider. Anticipating hearing from you soon.

**Reply by jennaswa** [reply]
Hi @Alentraca - Thank you for your interest in MDAnalysis! Please read our [GSoC 2024 blog post](https://www.mdanalysis.org/2024/02/27/gsoc2024/), which includes links to other important resources, for more details on how you can apply to work with MDAnalysis. The main thing is that the pre-proposal needs to be submitted by the end of the day (anywhere in the world) **tomorrow, March 15th**.

**Comment by Alentraca** [comment]
Hi @jennaswa 
Thanks for the timely response.
Loved the workshop and excited to contribute to MDAnalyis. Since been reviewing the [blogpost](https://www.mdanalysis.org/2024/02/27/gsoc2024/) , [wiki](https://github.com/MDAnalysis/mdanalysis/wiki/Google-Summer-Of-Code) and [Project5](https://github.com/MDAnalysis/mdanalysis/wiki/GSoC-2024-Project-Ideas#project-5-better-interfacing-of-blender-and-mdanalysis)
I have prepared my pre-proposal. 
It would be great to have remarks from MDAnalysis, @yuxuanzhuang , @BradyAJohnston with regards the Approach (in this proposal). Anticipating hearing from you soon

**Comment by kainszs** [comment]
Hi,

could you provide the original data for the [notebook](https://github.com/MDAnalysis/MDAnalysisWorkshop-Intro0.5Day/blob/feb24-ws/blender/Interactive_Blender_Notebook.ipynb)?

Best, Kai

**Reply by yuxuanzhuang** [reply]
The original data can be downloaded with
```python
import MDAnalysisData
MDAnalysisData.download(MDAnalysisData.adk_equilibrium.fetch_adk_equilibrium)
```
the original trajectory is IMO not important here though, you can use your own trajectories as well.
